### PR TITLE
Uncommented params from john.conf Options section.

### PR DESCRIPTION
**Pull request by jfoug**
Tied to #3598

### CODE REVIEW

File: run/john.conf [hunk 1]
```diff
@@ -42,9 +42,10 @@ SingleRules = Single
 # Default batch mode Wordlist rules
 BatchModeWordlistRules = Wordlist
 
-# Default wordlist mode rules when not in batch mode (if any)
-# If this is set and you want to run once without rules, use --rules:none
-#WordlistRules = Wordlist
+# Default wordlist mode rules when not in batch mode (if any).  If this is
+# changed from 'none' to have default rules applied, and you DO want to run
+# once without rules, use --rules:none on the command line.
+WordlistRules = none
```
Comment (magnumripper) [inline run/john.conf]: I think this might be a regression. We should compare using `-w` vs. `-w -ru:none` for some of the fastest formats.

Comment (jfoug) [reply run/john.conf]: Agreed.   I know the 'results' should be the same.  BUT there could be certain additional issues, going from 'no' rules, to a 'dummy' rule. 

I am glad question this showed up in the PR.  This change (and  the change to ``` DefaultIncrementalUTF8 = ASCII ``` ) are the 2 which troubled me the most.  Keeping the logic 'dictates' them, BUT they certainly give me the chills, thinking there is something not quite right.  The date formats, setting same logic for NULL as for "" is no big issue.  It is only at the 'exceptional event' of a log, and an extra if is no big deal.

Comment (magnumripper) [reply run/john.conf]: It should be easy to change the code to not enable rules at all for a null string.

Comment (jfoug) [reply run/john.conf]: That is probably a better solution, than using rules:none (much like the handling of date logic).  I will work on making it so, then it should be 'identical' both in logic and implementation, where with the patch as it stands it is only equal in logic.

Is the change to ``` DefaultIncrementalUTF8 = ASCII ``` OK?  That was about the only other thing that made the hair on the back of my neck stand up a bit.

Comment (magnumripper) [reply run/john.conf]: > Is the change to `DefaultIncrementalUTF8 = ASCII` OK? That was about the only other thing that made the hair on the back of my neck stand up a bit.

Yes, given that's the default it's actually better - because it's harder to miss!

**Comment by jfoug** [comment]
@magnumripper I believe that once the CI's complete, this is ready. 

It was tested using real wordlist, rules, no rules, etc.  The internal ST would not catch this.  The jtrts did catch an initial version which had problems.  But I rolled back changes, and simply detect when reading the config file that we have a "" string, and then set the variable to be NULL (i.e. original behavior).  Now, the jtrts.pl is happy, and all tests I could think of using rules were happy.

This should remove any possibility to a regression issue for this config option.

File: src/unicode.c [hunk 2]
```diff
@@ -971,7 +971,7 @@ int cp_name2id(char *encoding)
 	char enc[16] = "";
 	char *d = enc;
 
-	if (!encoding)
+	if (!encoding || !encoding[0])
 		return CP_UNDEF;
 	if (strlen(encoding) > sizeof(enc))
 		goto err;
```
Comment (magnumripper) [inline src/unicode.c]: What is the rationale for this change?

Comment (magnumripper) [reply src/unicode.c]: Never mind - I got the idea the change was the other way round. This makes sense!
